ouroboros-network-framework/src/Ouroboros/Network/Server2.hs

```diff
@@ -176,9 +174,24 @@ run ServerArguments {
             throwIO e
   where
 
-#if !defined(mingw32_HOST_OS)
     iseCONNABORTED :: IOError -> Bool
+#if defined(mingw32_HOST_OS)
+    -- On Windows the network packet classifies all errors
+    -- as OtherError. This means that we're forced to match
+    -- on the error string. The text string comes from
+    -- the network package's winSockErr.c, and if it ever
+    -- changes we must update our text string too.
+    iseCONNABORTED (IOError _ _ _ "Software caused connection abort (WSAECONNABORTED)" _ _) = True
+    iseCONNABORTED _ = False
+#else
```

Comment:
TODO: create ticket where we move eCONNABORTED detection to Snocket accept and push the osx workaround to network package.

Reply:
https://github.com/input-output-hk/ouroboros-network/issues/3475